### PR TITLE
Update to kubeflow 1.4 RC

### PR DESCRIPTION
**Pull request by ca-scribner**
* update image version to that from kubeflow 1.4 RC0
* update ports to match those used by new image

### CODE REVIEW

**Comment by ca-scribner** [comment]
Code itself should work, but tests are old and need updating.  Doing that next

**Comment by ca-scribner** [comment]
@DomFleischmann @knkski ok now ready for review.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+setenv =
+	PYTHONPATH={toxinidir}/src:{toxinidir}/lib
+deps =
+	-r requirements.txt
+
+[testenv:integration]
+deps =
+    {[testenv]deps}
+    -r requirements-integration.txt
+commands = pytest -v --tb native --show-capture=no --log-cli-level=INFO -s {posargs} {toxinidir}/tests/integration
```
[the remark this answers is not in the document]
Comment (DomFleischmann) [reply tox.ini:17]: I personally feel like different directories is a bit cleaner due to the way these tests are executed being so different. I feel it is also more intuitive for someone who is looking at the repository for the first time.

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+from pathlib import Path
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+log = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = "admission-webhook"
+CHARM_ROOT = "."
```
[the remark this answers is not in the document]
Comment (DomFleischmann) [reply tests/integration/test_charm.py:11]: This is not considered idiomatic for pytest-operator, I personally don't really feel its necessary either.

File: tox.ini (hunk 1 above)
[the remark this answers is not in the document]
Comment (DomFleischmann) [reply tox.ini:17]: The `--show-capture=no` I think is due to libjuju being very noisy.

File: tests/integration/test_charm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ca-scribner) [reply tests/integration/test_charm.py:11]: yeah I got into this habit in another charm that it actually made sense in.  I'll remove

[the remark this answers is not in the document]
Comment (ca-scribner) [reply tests/integration/test_charm.py:11]: Although in general, it feels more like a global config to me than something that should be hidden in a literal string in a single function

File: tox.ini (hunk 1 above)
[the remark this answers is not in the document]
Comment (ca-scribner) [reply tox.ini:17]: I prefer separate directories as well, although I use marks less than I should.  I think for most cases having a single top level `/tests` directory would work, but for some cases the extra resolution of the subdir would be nice.  I like having the same pattern for all our repos, so then the subdir feels like the better option.

Agreed with @DomFleischmann that `/tests/integration` feels more intuitive for someone new (it is easy to miss marks when browsing through files, but hard to miss directory structure).  I'd expect in future we'd also have support files (bundle yamls, maybe other things?) that would cleanly nest in a subdir too.

**Review by DomFleischmann — APPROVED** [review]

**Pull request merged**